### ISSUE TITLE
Mate 20 pro downgrade EMUI 9 failed. 

### ISSUE THREAD

**Issue by sharath91221** [ProfessorJTJ/HISuite-Proxy#145]
[logs.txt](https://github.com/ProfessorJTJ/HISuite-Proxy/files/7656490/logs.txt)
I have LYA-L09 C432 EMUI 11(11.0.0.172). I was able to downgrade to EMUI 10 or EMUI 10.1.  But I can't downgrade to EMUI 9 or EMUI 9.1. Hisuite proxy always throw "Apparently Firm(s) is/are not approved for installation and process will most likely fail in phone. You might want to consider cancelling it to avoid time waste." error. But I still continued and Hisuite was able to rollback but once its pushes the firmware to phone. The phone either throws "Software install failed get help from xxxxxx" or "Software install failed incompatibility with current version"

I tried recovery OS, rollback OS, force auth bridge. Still unable to downgrade to EMUI 9 or 9.1.   I bought harmony TP cable too. Attached the log file for reference. 
Currently, 
am on 10.0.0.154 version.
Tried Hisuite proxy 2.3.7 and 2.3.4 along with Hisuite 10.0.1.100_OVE and 11.0.0.510_OVE

**Comment by ProfessorJTJ** [comment]
Unlucky mate, this happens when Huawei doesn't approve an online rom for installation, so far we cannot trick it, I'll look further into it when I get time and I'll tell you if I find a solution for it

**Comment by ProfessorJTJ** [comment]
https://www.firmfinder.ml/327392 : LYA-LGRP2-OVS 9.1.0.347 - Approved

https://www.firmfinder.ml/182203 : LYA-L09-PRELOAD 9.0.0.16(C432R1) - Approved

https://www.firmfinder.ml/182204 : LYA-L09-CUST 9.0.0.10(C432) - Not Approved

The cust package is dropped by Huawei, I've ran a new search for old roms, I'll tell you if I find anything

**Comment by ProfessorJTJ** [comment]
Edit:

HiSuite is able to find a rollback package for you... you don't need to use HiSuite Proxy! ( Disable the proxy from HiSuite's settings, and try again )

313477 : LYA-L09-CUST 9.0.0.10(C432) : http://update.dbankcdn.com/download/data/pub_13/HWHOTA_hotaMigrate_900_9/91/v3/81ff8fc4cd6a4b949d278e71f0fc7a84/full/filelist.xml

**Comment by sharath91221** [comment]
Which version of Hisuite, you want to me to use for rollback without Hisuite proxy?

**Comment by sharath91221** [comment]
I uninstalled the patched hisuite and cleared "C:\Program Files (x86)\HiSuite" folder. Did a fresh installation of HiSuite_10.0.1.100_OVE@SilviuO.exe. Removed the hisuite app in my mobile. 

Hisuite without patching or Hisuite proxy gave me option to switch to other version and gave me option for 9.1.0.346. 
Currently, am in process of installation. I will keep you posted.

**Comment by sharath91221** [comment]
Am able to downgrade mate 20 pro LYA-L09 C432 from 11.0.0.172 to 9.1.346 successfully. @ProfessorJTJ Thank you so much 



Things I did. 
Using Hisuite proxy 2.3.7 with Hisuite 11.0.0.510, I downgraded the mobile from 11.0.0.172 to 10.0.0.154. 
During this process, as indicated in HiSuite proxy note: I lost the USB connection. So I bought a Harmony TP cable.
Now, I uninstalled uninstalled the patched Hisuite 11.0.0.510 and cleared "C:\Program Files (x86)\HiSuite" folder. Did a fresh installation of HiSuite_10.0.1.100_OVE.exe. Removed the hisuite app in my mobile.

Run a the HiSuite_10.0.1.100_OVE, it will ask to update application(do not update), just ignore it and connect your mobile and it will install a fresh copy of Hisuite app in the mobile and then enter the code and click on Update. It gives a option to SWITCH TO OTHER VERSION and provides 9.1.0.346 restore.
